pkg/client/client.go

```diff
@@ -39,6 +39,25 @@ type Client struct {
 }
 
 func NewClient(
+	p4RuntimeClient p4_v1.P4RuntimeClient,
+	deviceID uint64,
+	electionID p4_v1.Uint128,
+	optionsModifierFns ...func(*ClientOptions),
+) *Client {
+	options := defaultClientOptions
+	for _, fn := range optionsModifierFns {
+		fn(&options)
+	}
+	return &Client{
+		ClientOptions:   options,
+		P4RuntimeClient: p4RuntimeClient,
+		deviceID:        deviceID,
+		electionID:      electionID,
+		streamSendCh:    make(chan *p4_v1.StreamMessageRequest, 1000), // TODO: should be configurable
+	}
```

Comment:
this should just be `return NewClientForRole(p4RuntimeClient, deviceID, electionID, role, optionsModifierFns...)`

Reply:
Is it possible that you may still be looking at older commit? The latest commit I pushed has new function NewClientForRole that you proposed and the existing NewClient function is untouched to not break usage of this library for the current users. The NewClientForRole has the parameter 'role' that can be passed by the user to set the role param to the Client object.
It is not clear how we can 'return NewClientForRole(p4RuntimeClient, deviceID, electionID, role, optionsModifierFns...)' from existing NewClient function as 'role' is not available. Please correct me if I am mistaken.

Reply:
instead of duplicating the code, the implementation of `NewClient` should just be:
```go
func NewClient(
	p4RuntimeClient p4_v1.P4RuntimeClient,
	deviceID uint64,
	electionID p4_v1.Uint128,
	role *p4_v1.Role,
	optionsModifierFns ...func(*ClientOptions),
) *Client {
    return NewClientForRole(p4RuntimeClient, deviceID, electionID, nil, optionsModifierFns...)
}
```

Reply:
Ok. I thought you mentioned that NewClient should be left untouched as per your comment below.

"can we write NewClientForRole here, and leave the current NewClient untouched? In this case, it's an easy way not to break any consumers of this module."

My initial changes introduced Role parameter in the 'NewClient' and also modified the example use cases in cmd/ to adhere to changes for 'NewClient', but were rolled-back based on your feedback.

I can re-do the changes once you confirm. Thanks for your patience with this change.

Reply:
untouched like the interface doesn't change...

Reply:
I add an extra `role` parameter in the code snippet above, which may have confused you. I removed it.